pkg/pillar/cmd/zedkube/clusterstatus.go

```diff
@@ -0,0 +1,258 @@
+		log.Errorf("publishKubeConfigStatus: cluster token is not from configitme or encrypted")
+	}
+	// publish the cluster status for the kube container
+	ctx.pubEdgeNodeClusterStatus.Publish("global", status)
+}
+
+func decryptClusterToken(ctx *zedkubeContext) (string, error) {
+	if !ctx.clusterConfig.CipherToken.IsCipher {
+		return "", fmt.Errorf("decryptClusterToken: cluster token is not encrypted")
+	}
+
+	decryptAvailable := ctx.subControllerCert != nil && ctx.subEdgeNodeCert != nil
+	if !decryptAvailable {
+		return "", fmt.Errorf("decryptClusterToken: certificates are not available")
+	}
+	status, decBlock, err := cipher.GetCipherCredentials(
+		&cipher.DecryptCipherContext{
+			Log:                  log,
+			AgentName:            agentName,
+			AgentMetrics:         ctx.cipherMetrics,
+			PubSubControllerCert: ctx.subControllerCert,
+			PubSubEdgeNodeCert:   ctx.subEdgeNodeCert,
+		},
+		ctx.clusterConfig.CipherToken)
+	if ctx.pubCipherBlockStatus != nil {
+		err2 := ctx.pubCipherBlockStatus.Publish(status.Key(), status)
+		if err2 != nil {
+			return "", fmt.Errorf("decryptClusterToken: publish failed %v", err2)
+		}
+	}
+	if err != nil {
+		ctx.cipherMetrics.RecordFailure(log, types.DecryptFailed)
+		return "", fmt.Errorf("decryptClusterToken: failed to decrypt cluster token: %v", err)
+	}
+
+	err = ctx.cipherMetrics.Publish(log, ctx.pubCipherMetrics, "global")
+	if err != nil {
+		log.Errorf("decryptClusterToken: publish failed for cipher metrics: %v", err)
+		return "", fmt.Errorf("decryptClusterToken: failed to publish cipher metrics: %v", err)
+	}
+
+	return decBlock.ClusterToken, nil
+}
+
+func updateClusterIPReadiness(ctx *zedkubeContext) (changed bool) {
+	var ready bool
+	haveClusterIPConfig := ctx.clusterConfig.ClusterInterface != "" &&
+		ctx.clusterConfig.ClusterIPPrefix != nil
+	if haveClusterIPConfig {
+		for _, port := range ctx.deviceNetworkStatus.Ports {
+			if port.InvalidConfig || port.IfName == "" {
+				continue
+			}
+			if port.Logicallabel != ctx.clusterConfig.ClusterInterface {
+				continue
+			}
+			for _, addr := range port.AddrInfoList {
+				if addr.Addr.Equal(ctx.clusterConfig.ClusterIPPrefix.IP) {
+					ready = true
+					break
+				}
+			}
+			if ready {
+				break
+			}
+		}
+	}
+	if ctx.clusterIPIsReady != ready {
+		ctx.clusterIPIsReady = ready
+		return true
+	}
+	return false
+}
+
+func startClusterStatusServer(ctx *zedkubeContext) {
+	if ctx.statusServer != nil {
+		// Already running.
+		return
+	}
+	mux := http.NewServeMux()
+	mux.HandleFunc("/status", func(w http.ResponseWriter, r *http.Request) {
+		clusterStatusHTTPHandler(w, r, ctx)
+	})
+	ctx.statusServer = &http.Server{
+		Addr:    ctx.clusterConfig.ClusterIPPrefix.IP.String() + ":" + types.ClusterStatusPort,
+		Handler: mux,
+	}
+	ctx.statusServerWG.Add(1)
+
+	// Start the server in a goroutine
+	go func() {
+		defer ctx.statusServerWG.Done()
+		if err := ctx.statusServer.ListenAndServe(); err != nil && err != http.ErrServerClosed {
```

Comment:
This listens on all IP addresses. What's the security approach for this open port? (And which port number does it run on? Some documentation seems to be missing ;-)

Reply:
This actually is only listen on the clusterprefix IP address with the cluster status port.
netstat showing this service:
```
tcp        0      0 10.244.244.2:12346      0.0.0.0:*               LISTEN      4193/zedkube
```
this is documented in pkg/pillar/doc/zedkube.md in this PR. Updated the above with comments to that.

Reply:
Is this clusterprefix IP address reachable from the outside of the cluster?

Reply:
It should not be reachable from outside of the local network, the router will not know this private ip prefix.

Reply:
@eriknordmark Would you be OK with this approach? When this is merged I can open a followup PR with networking changes and the ACL rule allowing access to this port added.